src/xrpld/app/tx/detail/Payment.cpp

```diff
@@ -341,6 +341,28 @@ Payment::doApply()
 
     AccountID const dstAccountID(ctx_.tx.getAccountID(sfDestination));
     STAmount const dstAmount(ctx_.tx.getFieldAmount(sfAmount));
+
+    if (ctx_.isDelegated && !ctx_.gpSet.empty())
+    {
+        // If gpSet is not empty, granular delegation is happening.
+        // Currently we only support PaymentMint and PaymentBurn granular
+        // permission. PaymentMint means the sender is the issuer. PaymentBurn
+        // means the destination is the issuer.
+        bool authorized = false;
+        auto const amountIssue = dstAmount.issue();
+        if (isXRP(amountIssue))
+            return tecNO_AUTH;
+        if (amountIssue.account == account_ &&
```

Comment:
Can simplify here and below:

    if (amountIssue.account == account_ &&
            ctx_.gpSet.contains(gpPaymentMint))